### DISCUSSION TITLE
Connect timed out Error

### DISCUSSION THREAD

**Discussion by AbdullahGaw** [aws/aws-iot-device-sdk-python#320] [Q&A] [answered]
Hi, I am trying to connect my Digi device to the MQTT broker and I keep getting the error below can anybody help please

Connect timed out
Traceback (most recent call last):
  File "/etc/config/scripts/awsmqttlient.py", line 22, in <module>
    myMQTTClient.connect()
  File "/etc/config/scripts/lib/python3.10/site-packages/AWSIoTPythonSDK/MQTTLib.py", line 520, in connect
    return self._mqtt_core.connect(keepAliveIntervalSecond)
  File "/etc/config/scripts/lib/python3.10/site-packages/AWSIoTPythonSDK/core/protocol/mqtt_core.py", line 199, in connect
    raise connectTimeoutException()
AWSIoTPythonSDK.exception.AWSIoTExceptions.connectTimeoutException

**Comment by jmklix** [comment, marked as answer]
I would recommend trying to use the [v2 sdk](https://github.com/aws/aws-iot-device-sdk-python-v2) over this one.

For your error I would make sure that your certificates and IoT policies are set up correctly.

**Reply by AbdullahGaw** [reply]
Thank you I will look into that.

**Reply by AbdullahGaw** [reply]
I think you are right my certificates are messed up. I found that I do not have a root-CA.crt How do i get one of those ? it does not come with AWS connection kit or even when i create a certificate though (security/certificates) it does not appear to be the same as the root-CA.

**Reply by jmklix** [reply]
You can get the amazon root-CA.crt from [here](https://www.amazontrust.com/repository/). You can also download it with curl:
```
curl https://www.amazontrust.com/repository/AmazonRootCA1.pem > root-CA.crt
```

**Reply by AbdullahGaw** [reply]
Thank you it is working now
